lib/identity_cache/with_primary_index.rb

```diff
@@ -0,0 +1,203 @@
+# frozen_string_literal: true
+module IdentityCache
+  module WithPrimaryIndex
+    extend ActiveSupport::Concern
+
+    include WithoutPrimaryIndex
```

Comment:
I think this approach makes sense from a reusability standpoint, but I don't like that `IdentityCache` modules include both `WithPrimaryIndex` and `WithoutPrimaryIndex`. That said, I do agree with the creation of this module. 

We should work toward having `WithPrimaryIndex` and `WithoutPrimaryIndex` being our base includes and having `include IdentityCache` be an alias for `include IdentityCache::WithPrimaryIndex` for legacy reasons. In the meantime, this is still better than the previous code, so I'll approve. 👍

Reply:
I agree that the `WithoutPrimaryIndex` name is kind of weird, once `WithPrimaryIndex` also gets included.  Maybe `WithoutPrimaryIndex` should be renamed to `Base` in the future so it isn't implying something with the name that could become false by including another module.

I would like to avoid `include IdentityCache` in the future, since that also includes its constants into the module.  I think it would be better to use `include IdentityCache::Base` and then call `cache_primary_index` to opt-in to using the primary index, which could internally include a module to make primary index methods available.  Calling a method to opt-in to the primary index would also allow us to pass in options, like explicitly specifying the embedded associations in one place, so that it is clear how much we are embedding.